pkg/agent/controller/egress/egress_controller.go

```diff
@@ -468,12 +481,51 @@ func (c *EgressController) unrealizeEgressIP(egressName, egressIP string) error
 			}
 			ipState.flowsInstalled = false
 		}
-		c.idAllocator.release(ipState.mark)
+		c.markIdAllocator.release(ipState.mark)
 	}
 	delete(c.egressIPStates, egressIP)
 	return nil
 }
 
+// realizeEgressMeter realizes an Egress Meter id and add meter rate limit rule.
+func (c *EgressController) realizeEgressMeter(egressName string, bandwidth uint32, eState *egressState) error {
+	c.egressStatesMutex.Lock()
+	defer c.egressStatesMutex.Unlock()
+	var err error
+	if eState.bandwidth == 0 {
+		eState.meterId, err = c.meterIdAllocator.allocate()
+		if err != nil {
+			return fmt.Errorf("error allocating meter id for Egress %s: %v", egressName, err)
+		}
+	}
+	if err := c.ofClient.AddMeterFlow(eState.meterId, bandwidth); err != nil {
+		return err
+	}
+	// Since the bandwidth has been updated, need to add SNAT flows for pods again, so clean the EgressState ofPorts
+	// cache.
+	eState.ofPorts = sets.NewInt32()
+	eState.bandwidth = bandwidth
+	return nil
+}
+
+// unrealizeEgressMeter unrealizes an Egress Meter id and reset egressState meter info.
+func (c *EgressController) unrealizeEgressMeter(meterId uint32, eState *egressState) error {
```

Comment:
Probably call it "deleteEgressMeter".

Reply:
Should we call DeleteMeterFlow() in this func?

Reply:
Thanks for the review, I have optimized the `unrealizeEgressMeter ` process, and combined `DeleteMeterFlow ` into it.